UPGRADE.asciidoc

```diff
@@ -1781,15 +1781,24 @@ The api_user table has been deprecated
 
 Any users in that were in the api_user table should be migrated to PacketFence local account (password table)
 
-Database schema update (All linux distributions)
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Database schema and privileges update (All linux distributions)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Changes have been made to the database schema. You will need to update it accordingly.
-An SQL upgrade script has been provided to upgrade the database from the X.X schema to X.Y.
+Changes have been made to the database schema and to the backup script
+`/usr/local/pf/addons/database-backup-and-maintenance.sh`. Starting from now,
+we will dump stored routines with the PacketFence database. You will need to update:
 
-To upgrade the database schema, run the following command:
+ * database schema
+ * privileges of the user configured in PacketFence (so he can dump stored routines)
+
+An SQL upgrade script has been provided to upgrade the database from the X.X schema to X.Y.
+To upgrade the database schema and privileges, run the following commands:
 
-  mysql -u root -p pf -v < /usr/local/pf/db/upgrade-X.X.X-X.Y.Z.sql
+ PF_USER=$(perl -I/usr/local/pf/lib -Mpf::db -e 'print $pf::db::DB_Config->{user}')
+ mysql -u root -p -e "GRANT SELECT ON mysql.proc TO $PF_USER@'%';" \
+ -e "GRANT SELECT ON mysql.proc TO $PF_USER@'localhost';" \
+ -e "FLUSH PRIVILEGES;"
+ mysql -u root -p pf -v < /usr/local/pf/db/upgrade-X.X.X-X.Y.Z.sql
```

Comment:
The instructions updating the privileges should be separate from the database instructions.
Provide a bash script to perform the upgrade.
Add a step to run the permission script across the entire cluster since the mysql system tables are not replicated.

Reply:
> Add a step to run the permission script across the entire cluster since the mysql system tables are not replicated

I agree but is it really necessary ? In a cluster env, we use a dedicated account (`pfcluster`) to take dumps. This account is created by hand with correct rights.

Reply:
As long as that user is always created with the proper permissions you are right that step is not needed.
Just wrap those steps into a bash script then we are good.

Reply:
As long as that user is always created with the proper permissions.
Then you are right that step is not needed.
Just wrap those steps into a bash script then we are good.

Reply:
I apply my one-liner on a host in a Galera Cluster: grants were replicated on all hosts.

Reply:
Then we are good

Reply:
I will prepare a shell script and split instructions in two sections.